cobalt/persistent_storage/persistent_settings_test.cc

```diff
@@ -547,22 +548,18 @@ TEST_F(PersistentSettingTest, InvalidSettings) {
   base::OnceClosure closure = base::BindOnce(
       [](PersistentSettings* persistent_settings,
          base::WaitableEvent* test_done) {
-        ASSERT_TRUE(
+        EXPECT_TRUE(
             persistent_settings->GetPersistentSettingAsBool("key", true));
-        ASSERT_TRUE(
+        EXPECT_TRUE(
             persistent_settings->GetPersistentSettingAsBool("key", false));
         test_done->Signal();
       },
       persistent_settings.get(), &test_done_);
   persistent_settings->SetPersistentSetting(
-      "key", std::make_unique<base::Value>(true), std::move(closure));
+      "key", std::make_unique<base::Value>(true), std::move(closure), true);
   test_done_.Wait();
   test_done_.Reset();
 
-  // Sleep for one second to allow for the previous persistent_setting's
-  // JsonPrefStore instance time to write to disk before creating a new
-  // persistent_settings and JsonPrefStore instance.
-  base::PlatformThread::Sleep(base::TimeDelta::FromSeconds(1));
```

Comment:
We originally used the sleep() function to wait for the previous test to finish. However, this might not a good practice for unit testing. Unit tests should be independent of each other, so that they can be run in any order. A better approach would be to break this function into a few smaller tests.

Reply:
So splitting this test up isnt that trivial.

1st pair of asserts: tests that persistent settings are not set
2nd pair of asserts: tests that persistent settings are set

3rd pair of asserts: tests that restarted/invalid persistent settings are set
4th pair of asserts: tests that restarted/invalid persistent settings are not set

5th pair of asserts: tests that restarted/invalid persistent settings are reset

The current teardowns that happen inbetween tests will reset all of this as the original goal was to keep each test standalone.

I think if we want to change that we should do so in another CL and have a discussion about that separately.

Reply:
I agree that splitting this up into smaller tests is good, but also agree that this could be done in a follow-up PR, as this current one already should improve status quo.

In the follow-up, you can move the pre-conditions setup part into the test fixture itself, something like
```
PersistentSettingTest2 {
   void Setup() { ... 
      persistent_settings = std::make_unique<PersistentSettings>(kPersistentSettingsJson);
   } 
   void make_some_test_settings(foo) {
     persistent_settings->SetPersistentSetting(foo)
   }
}

TEST_F(PersistentSettingTest2) {
   make_some_test_settings("foo");
   ASSERT_TRUE(
            persistent_settings->GetPersistentSettingAsBool("foo", false));
}
```
It consolidates setup code and makes tests shorter / easier to extend.